--- a/src/arm/api.h	2026-03-19 03:09:33.516779300 -0400
+++ b/src/arm/api.h	2026-03-19 03:10:24.246565400 -0400
@@ -1,3 +1,10 @@
+/* MSVC does not support C99 [restrict static N] array qualifiers */
+#ifdef _MSC_VER
+#define CPUINFO_RESTRICT_STATIC
+#else
+#define CPUINFO_RESTRICT_STATIC restrict static
+#endif
+
 #pragma once
 
 #include <stdbool.h>
@@ -79,34 +86,34 @@
 
 #ifndef __cplusplus
 	CPUINFO_INTERNAL void cpuinfo_arm_chipset_to_string(
-		const struct cpuinfo_arm_chipset chipset[restrict static 1],
-		char name[restrict static CPUINFO_ARM_CHIPSET_NAME_MAX]);
+		const struct cpuinfo_arm_chipset chipset[CPUINFO_RESTRICT_STATIC 1],
+		char name[CPUINFO_RESTRICT_STATIC CPUINFO_ARM_CHIPSET_NAME_MAX]);
 
 	CPUINFO_INTERNAL void cpuinfo_arm_fixup_chipset(
-		struct cpuinfo_arm_chipset chipset[restrict static 1], uint32_t cores, uint32_t max_cpu_freq_max);
+		struct cpuinfo_arm_chipset chipset[CPUINFO_RESTRICT_STATIC 1], uint32_t cores, uint32_t max_cpu_freq_max);
 
 	CPUINFO_INTERNAL void cpuinfo_arm_decode_vendor_uarch(
 		uint32_t midr,
 	#if CPUINFO_ARCH_ARM
 		bool has_vfpv4,
 	#endif
-		enum cpuinfo_vendor vendor[restrict static 1],
-		enum cpuinfo_uarch uarch[restrict static 1]);
+		enum cpuinfo_vendor vendor[CPUINFO_RESTRICT_STATIC 1],
+		enum cpuinfo_uarch uarch[CPUINFO_RESTRICT_STATIC 1]);
 
 	CPUINFO_INTERNAL void cpuinfo_arm_decode_cache(
 		enum cpuinfo_uarch uarch,
 		uint32_t cluster_cores,
 		uint32_t midr,
-		const struct cpuinfo_arm_chipset chipset[restrict static 1],
+		const struct cpuinfo_arm_chipset chipset[CPUINFO_RESTRICT_STATIC 1],
 		uint32_t cluster_id,
 		uint32_t arch_version,
-		struct cpuinfo_cache l1i[restrict static 1],
-		struct cpuinfo_cache l1d[restrict static 1],
-		struct cpuinfo_cache l2[restrict static 1],
-		struct cpuinfo_cache l3[restrict static 1]);
+		struct cpuinfo_cache l1i[CPUINFO_RESTRICT_STATIC 1],
+		struct cpuinfo_cache l1d[CPUINFO_RESTRICT_STATIC 1],
+		struct cpuinfo_cache l2[CPUINFO_RESTRICT_STATIC 1],
+		struct cpuinfo_cache l3[CPUINFO_RESTRICT_STATIC 1]);
 
 	CPUINFO_INTERNAL uint32_t cpuinfo_arm_compute_max_cache_size(
-		const struct cpuinfo_processor processor[restrict static 1]);
+		const struct cpuinfo_processor processor[CPUINFO_RESTRICT_STATIC 1]);
 #else /* defined(__cplusplus) */
 	CPUINFO_INTERNAL void cpuinfo_arm_decode_cache(
 		enum cpuinfo_uarch uarch,
