--- a/src/arm/uarch.c	2026-03-19 03:09:33.755566100 -0400
+++ b/src/arm/uarch.c	2026-03-19 03:09:47.849262700 -0400
@@ -10,8 +10,8 @@
 #if CPUINFO_ARCH_ARM
 	bool has_vfpv4,
 #endif /* CPUINFO_ARCH_ARM */
-	enum cpuinfo_vendor vendor[restrict static 1],
-	enum cpuinfo_uarch uarch[restrict static 1])
+	enum cpuinfo_vendor vendor[CPUINFO_RESTRICT_STATIC 1],
+	enum cpuinfo_uarch uarch[CPUINFO_RESTRICT_STATIC 1])
 {
 	switch (midr_get_implementer(midr)) {
 		case 'A':
