--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -16224,6 +16224,11 @@ config_setting(
     values = {"cpu": "x64_windows"},
 )

+config_setting(
+    name = "windows_arm64",
+    values = {"cpu": "arm64_windows"},
+)
+
 config_setting(
     name = "windows_x86_64_clang",
     values = {
@@ -16378,6 +16383,7 @@ selects.config_setting_group(
     name = "aarch64",
     match_any = [
         ":android_arm64",
+        ":windows_arm64",
         ":ios_arm64",
         ":ios_arm64e",
         ":ios_sim_arm64",
