--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -16224,6 +16224,11 @@ config_setting(
     values = {"cpu": "x64_windows"},
 )

+config_setting(
+    name = "windows_arm64",
+    values = {"cpu": "arm64_windows"},
+)
+
 config_setting(
     name = "windows_x86_64_clang",
     values = {
@@ -16378,6 +16383,7 @@ selects.config_setting_group(
     name = "aarch64",
     match_any = [
         ":android_arm64",
+        ":windows_arm64",
         ":ios_arm64",
         ":ios_arm64e",
         ":ios_sim_arm64",
--- a/build_defs.bzl
+++ b/build_defs.bzl
@@ -146,7 +146,7 @@ def xnnpack_cc_library(
             ":emscripten_wasm": wasm_srcs,
             ":emscripten_wasmsimd": wasmsimd_srcs,
             ":emscripten_wasmrelaxedsimd": wasmrelaxedsimd_srcs,
-            "//conditions:default": [],
+            "//conditions:default": x86_srcs,
         }),
         copts = [
             "-Iinclude",
@@ -179,7 +179,7 @@ def xnnpack_cc_library(
             ":emscripten_wasm": wasm_copts,
             ":emscripten_wasmsimd": wasmsimd_copts,
             ":emscripten_wasmrelaxedsimd": wasmrelaxedsimd_copts,
-            "//conditions:default": [],
+            "//conditions:default": gcc_x86_copts,
         }) + select({
             ":windows_x86_64_clang": ["/clang:" + opt for opt in gcc_copts],
             ":windows_x86_64_mingw": gcc_copts,
@@ -249,6 +249,7 @@ def xnnpack_aggregate_library(
             ":emscripten_wasmsimd": wasmsimd_deps,
             ":emscripten_wasmrelaxedsimd": wasmrelaxedsimd_deps,
             ":riscv": riscv_deps,
+            "//conditions:default": x86_deps,
         }),
         defines = defines,
         compatible_with = compatible_with,
